src/wamp-client-fs.ts

```diff
@@ -29,85 +29,74 @@ export class WampClientFileSystem implements FileSystem {
                         res => this.events.emit(fsEvent, res && res[0])
                     )
                 });
-                return resolve(this)
+                resolve(this)
             };
         }), `Cant't open connection to the WAMP server at ${baseUrl} for ${initTimeout}ms.`);
     }
 
-    saveFile(fullPath:string, newContent:string): Promise<void> {
-        return new Promise<void>((resolve, reject) => {
-            if (!this.session || !this.session.isOpen) {
-                return reject(noConnectionError);
-            }
-            this.session.call(`${this.realmPrefix}saveFile`, [fullPath, newContent])
-                .then(() => resolve())
-                .catch(error => reject(new Error(error)))
-        });
+    async saveFile(fullPath:string, newContent:string): Promise<void> {
+        this.throwIfDisconnected();
+
+        try {
+            return await this.session.call<void>(`${this.realmPrefix}saveFile`, [fullPath, newContent]);
+        } catch (error) {
+            throw new Error(error);
```

Comment:
this wrapping with an `Error` seems very wrong to me. are you 100% certain that `error` is a string?